src/main/java/seedu/address/model/person/MatricNumber.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's matricNumber in the club manager.
+ * Guarantees: immutable; is valid as declared in {@link #isValidMatricNumber(String)}
+ */
+public class MatricNumber {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "MatricNumber should only contain alphanumeric characters, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum} ]*";
```

Comment:
Can constrain this abit more to fit actual matric no length and format?

Reply:
if we making this application purely for NUS then it is ok, but if we want other universities and schools to use, it would be more difficult to restrict more, because matric number have different length and format for other schools

Reply:
it's better that we make the scope more narrow for now as stated in the project requirement

Reply:
ok no problem, i will do it :)